tests/Aql/SubqueryEndExecutorTest.cpp

```diff
@@ -0,0 +1,295 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2019 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Michael Hackstein
+/// @author Markus Pfeiffer
+////////////////////////////////////////////////////////////////////////////////
+
+#include "AqlItemBlockHelper.h"
+#include "RowFetcherHelper.h"
+#include "gtest/gtest.h"
+
+#include "Aql/OutputAqlItemRow.h"
+#include "Aql/Stats.h"
+#include "Aql/SubqueryEndExecutor.h"
+
+#include "Logger/LogMacros.h"
+
+#include "Basics/VelocyPackHelper.h"
+
+using namespace arangodb;
+using namespace arangodb::aql;
+using namespace arangodb::tests;
+using namespace arangodb::tests::aql;
+using namespace arangodb::basics;
+
+namespace {
+SubqueryEndExecutorInfos MakeBaseInfos() {
+  // one input register (RegisterId 0) and one output register (RegisterId 1)
+  // size_t regs = 2;
+
+  auto inputRegisters = std::make_shared<std::unordered_set<RegisterId>>(
+      std::initializer_list<RegisterId>{0});
+
+  auto outputRegisters = std::make_shared<std::unordered_set<RegisterId>>(
+      std::initializer_list<RegisterId>({0}));
+
+  return SubqueryEndExecutorInfos(inputRegisters, outputRegisters, 1, 1, {}, {},
+                                  nullptr, RegisterId{0});
+}
+
+}  // namespace
+
+class SubqueryEndExecutorTest : public ::testing::Test {
+public:
+  SubqueryEndExecutorTest() : infos(MakeBaseInfos()) {};
+ protected:
+  ResourceMonitor monitor;
+  AqlItemBlockManager itemBlockManager{&monitor, SerializationFormat::SHADOWROWS};
+  SubqueryEndExecutorInfos infos;
+
+
+  void ExpectedValues(OutputAqlItemRow& itemRow,
+                      const std::vector<std::string> expectedStrings,
```

Comment:
```suggestion
                      std::vector<std::string> const& expectedStrings,
```

we can save a copy here for performance in test execution ;)